google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/project/ProjectSelector.java

```diff
@@ -64,9 +69,32 @@
   // null by default. set by caller to allow project selector to pre-select active project.
   private Project ideProject;
 
+  @VisibleForTesting GoogleLoginListener googleLoginListener;
+
   /** @param ideProject IDE {@link Project} to be used to update active cloud project settings. */
   public ProjectSelector(@Nullable Project ideProject) {
     this.ideProject = ideProject;
+
+    // listen to login changes to show sign in prompt when no one is signed in.
+    googleLoginListener =
+        () -> {
+          if (cloudProject != null) {
+            ApplicationManager.getApplication()
+                .invokeLater(
+                    () -> {
+                      updateCloudProjectSelection(cloudProject);
+                      notifyProjectSelectionListeners();
+                    });
+          }
+        };
+    Stream.of(ProjectManager.getInstance().getOpenProjects())
+        .forEach(
+            project -> {
+              project
+                  .getMessageBus()
+                  .connect()
```

Comment:
should we pass in a disposable so that the subscriber get detached when closed?

Reply:
I tried to think what would be that disposable in case of universal component and couldn't find a solution that wouldn't fail - i.e. we cannot count on being inside of the dialog or tie to the IDE project - those could be flaky. Open to suggestions or hints.

Reply:
hmm I see. for the deployment window as an example, each time its opened a new project selector instance is created. So if we don't pass in a disposable, then the subscribers will never get killed and will be stuck orphaned subscribers - does that sound correct?

Reply:
Sure, we need to get rid of that listener. It seems like a new case, I don't see it in our code yet - do you remember if it's possible to unregister a message bus listener?

Reply:
`MessageBusConnection` has a `disconnect` method. That can be called. Its likely that I have also made this error previously in our code (not passing in a disposable or explicitly disconnecting). We can (and should) fix this in a separate pr. WDYT?

Reply:
Sounds good, let's do the disconnect - I think we can count on an UI component being removed from the container event to do that cleanly. I'll file a separate PR for that.